src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitedArray = date.split(fromFormat[fromFormat.length - 1]);
+  let year = '';
+  const separator = toFormat[toFormat.length - 1];
+
+  toFormat.pop();
+
+  for (const i of toFormat) {
+    if (i === 'DD') {
+      toFormat[toFormat.indexOf(i)] = splitedArray[fromFormat.indexOf(i)];
+    }
+
+    if (i === 'MM') {
+      toFormat[toFormat.indexOf(i)] = splitedArray[fromFormat.indexOf(i)];
+    }
+
+    if ((i === 'YYYY' || i === 'YY') && fromFormat.includes(i)) {
+      year = splitedArray[fromFormat.indexOf(i)];
+      toFormat[toFormat.indexOf(i)] = year;
+    } else if (i === 'YY') {
+      year = splitedArray[fromFormat.indexOf('YYYY')].slice(2);
```

Comment:
When slicing the year to get the last two digits, ensure that the 'YYYY' format is present in the `fromFormat` array to avoid potential errors.